drools-wb-screens/drools-wb-guided-dtable-editor/drools-wb-guided-dtable-editor-client/src/main/java/org/drools/workbench/screens/guided/dtable/client/widget/table/GuidedDecisionTableModellerPresenter.java

```diff
@@ -0,0 +1,472 @@
+            return true;
+        }
+        return dtPresenter.getAccess().isEditable();
+    }
+
+    @Override
+    public GuidedDecisionTableModellerView getView() {
+        return view;
+    }
+
+    @Override
+    public void onInsertColumn() {
+        final GuidedDecisionTableView.Presenter dtPresenter = getActiveDecisionTable();
+        if ( dtPresenter == null ) {
+            return;
+        }
+        view.onInsertColumn();
+    }
+
+    @Override
+    public void setZoom( final int zoom ) {
+        view.setZoom( zoom );
+    }
+
+    @Override
+    public void onDecisionTableSelected( final @Observes DecisionTableSelectedEvent event ) {
+        final GuidedDecisionTableView.Presenter dtPresenter = event.getPresenter();
+        if ( dtPresenter == null ) {
+            return;
+        }
+        if ( !isDecisionTableAvailable( dtPresenter ) ) {
+            return;
+        }
+        if ( dtPresenter.equals( activeDecisionTable ) ) {
+            return;
+        }
+        doDecisionTableSelected( dtPresenter );
+    }
+
+    private void doDecisionTableSelected( final GuidedDecisionTableView.Presenter dtPresenter ) {
+        //Store selected decision table
+        activeDecisionTable = dtPresenter;
+
+        //Bootstrap Decision Table analysis
+        for ( GuidedDecisionTableView.Presenter p : availableDecisionTables ) {
+            if ( p.equals( dtPresenter ) ) {
+                p.initialiseAnalysis();
+            } else {
+                p.terminateAnalysis();
+            }
+        }
+
+        //Update view with selected decision table detail
+        final GuidedDecisionTable52 model = dtPresenter.getModel();
+        dtPresenter.getPackageParentRuleNames( new ParameterizedCommand<Collection<String>>() {
+            @Override
+            public void execute( Collection<String> availableParentRuleNames ) {
+                view.refreshRuleInheritance( model.getParentName(),
+                                             availableParentRuleNames );
+            }
+        } );
+        view.setEnableColumnCreation( dtPresenter.getAccess().isEditable() );
+        view.refreshAttributeWidget( model.getAttributeCols() );
+        view.refreshMetaDataWidget( model.getMetadataCols() );
+        view.refreshConditionsWidget( model.getConditions() );
+        view.refreshActionsWidget( model.getActionCols() );
+        view.refreshColumnsNote( dtPresenter.hasColumnDefinitions() );
+
+        //Delegate highlighting of selected decision table to ISelectionManager
+        view.select( dtPresenter.getView() );
+    }
+
+    @Override
+    public void onDecisionTableLinkedColumnSelected( final @Observes DecisionTableColumnSelectedEvent event ) {
+        if ( event.getColumn() == null ) {
+            return;
+        }
+        view.selectLinkedColumn( event.getColumn() );
+    }
+
+    @Override
+    public void onRefreshAttributesPanelEvent( final @Observes RefreshAttributesPanelEvent event ) {
```

Comment:
After much fiddling around I found out that ```Consumer``` is not available in GWT 2.8.0-beta1 but I have removed the duplicate code and changed to use a lambda instead.